### PR TITLE
Backend: KuudraAPI

### PR DESCRIPTION
**Pull request by ItsEmpa**
## What
Adds a KuudraAPI, and the events KuudraEnterEvent and KuudraCompleteEvent

Needed for #1190 

## Changelog Technical Details
+ Added KuudraAPI. - Empa
+ Added KuudraEnterEvent and KuudraCompleteEvent. - Empa

### CODE REVIEW

File: src/main/java/at/hannibal2/skyhanni/features/nether/KuudraAPI.kt [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+
+    private val tierPattern by scoreboardGroup.pattern(
+        "tier",
+        " §7⏣ §cKuudra's Hollow §8\\(T(?<tier>.*)\\)"
+    )
+
+    private val completePattern by chatGroup.pattern(
+        "complete",
+        "§.\\s*(?:§.)*KUUDRA DOWN!"
+    )
```
Comment (CalMWolfs) [inline src/main/java/at/hannibal2/skyhanni/features/nether/KuudraAPI.kt]:
```suggestion
    private val patternGroup = RepoPattern.group("data.kuudra")

    private val tierPattern by patternGroup.pattern(
        "scoreboard.tier",
        " §7⏣ §cKuudra's Hollow §8\\(T(?<tier>.*)\\)"
    )
    private val completePattern by patternGroup.pattern(
        "chat.complete",
        "§.\\s*(?:§.)*KUUDRA DOWN!"
    )
```
Seeing as there is only two patterns may as well do something like this. Also this is made on web so no clue if it compiles though it should

File: src/main/java/at/hannibal2/skyhanni/features/nether/KuudraAPI.kt [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+        val location: LorenzVec?,
+        val tierNumber: Int,
+        var doneToday: Boolean = false,
+    ) {
+        fun getDisplayName() = "Tier $tierNumber ($name)"
+    }
+
+}
+
+
```
Comment (CalMWolfs) [inline src/main/java/at/hannibal2/skyhanni/features/nether/KuudraAPI.kt]:
```suggestion
}

```
remove extra lines

**Review by CalMWolfs — CHANGES_REQUESTED** [review]
just these two smaller changes i think

**Comment by ItsEmpa** [comment]
done

**Review by CalMWolfs — APPROVED** [review]

**Review by hannibal002 — APPROVED** [review]

**Pull request merged**